base/MAPL_Resource.F90

```diff
@@ -0,0 +1,411 @@
+      labels_with_prefix(2) = trim(component_type)//"_"//trim(label)
+      labels_with_prefix(3) = trim(label)
+      labels_with_prefix(4) = trim(compname)//MAPL_CF_COMPONENT_SEPARATOR//trim(label)
+
+   end function get_labels_with_prefix
+
+   ! If possible, find label or label with prefix. Out: logical to if label found, 
+   ! version of label found,
+   subroutine get_label_to_use(config, label, compname, label_is_present, label_to_use, rc)
+      type(ESMF_Config), intent(inout) :: config
+      character(len=*), intent(in) :: label
+      character(len=*), intent(in) :: compname
+      logical, intent(out) :: label_is_present
+      character(len=:), allocatable, intent(out) :: label_to_use
+      integer, optional, intent(out) :: rc
+
+      character(len=ESMF_MAXSTR), allocatable :: labels_to_try(:)
+      integer :: i
+      integer :: status
+
+      label_is_present = .false.
+      labels_to_try = get_labels_with_prefix(compname, label)
+
+      do i = 1, size(labels_to_try)
+         label_to_use = trim(labels_to_try(i))
+         call ESMF_ConfigFindLabel(config, label = label_to_use, isPresent = label_is_present, _RC)
+
+         if (label_is_present) then
+            exit
+         end if
+      end do
+
+      _RETURN(_SUCCESS)
+
+   end subroutine get_label_to_use
+
+   ! Find value of scalar variable in config
+   subroutine MAPL_GetResource_config_scalar(config, val, label_to_find, default, compname, rc)
+      type(ESMF_Config), intent(inout) :: config
+      class(*), intent(inout) :: val
+      character(len=*), intent(in) :: label_to_find
+      class(*), optional, intent(in) :: default
+      character(len=*), optional, intent(in) :: compname
+      integer, optional, intent(out) :: rc
+
+      integer :: status, printrc
+      logical :: default_is_present, label_is_present
+      character(len=:), allocatable :: label_to_print
+      character(len=:), allocatable :: label
+
+      default_is_present = present(default)
+
+      if (default_is_present) then
+         _ASSERT(same_type_as(val, default), "Value and default must have same type")
+      end if
+
+      ! If compname is present, find label in some form in config. Else search
+      ! for exact label
+      if (present(compname)) then
+         call get_label_to_use(config, label_to_find, compname, label_is_present, label, _RC)
+      else
+         label = label_to_find
+         call ESMF_ConfigFindLabel(config, label = label, isPresent = label_is_present, _RC)
+      end if
+
+      ! No default and not in config, error
+      ! label or default must be present
+      if (.not. label_is_present .and. .not. default_is_present) then
+         if (present(rc)) rc = ESMF_FAILURE
+         return
+      end if
+
+      select type(val)
+      type is(integer(int32))
+         if (default_is_present .and. .not. label_is_present) then
+            select type(default)
+            type is(integer(int32))
+               val = default
+            end select
```

Comment:
There should be a check that val and default have the same type.   Unfortunately the handy intrinsic `SAME_TYPE_AS()` does not quite do what we want (long story of confusion in the standard committee), so you may need a DEFAULT block in each of these.  Ugh.

Reply:
Done.